django/db/backends/oracle/base.py

```diff
@@ -49,9 +49,12 @@ def _setup_environment(environ):
 
 
 try:
-    import cx_Oracle as Database
+    import oracledb as Database
 except ImportError as e:
-    raise ImproperlyConfigured("Error loading cx_Oracle module: %s" % e)
+    try:
+        import cx_Oracle as Database
```

Comment:
This should raise a deprecation warning, see [comment](https://code.djangoproject.com/ticket/33817#comment:1) and [docs about deprecating a feature](https://docs.djangoproject.com/en/4.0/internals/contributing/writing-code/submitting-patches/#deprecating-a-feature).

Reply:
Deprecation should be documented, please see guide that I mentioned :point_up:

Reply:
Tests of deprecation warnings are missing.

Reply:
Hi, I've updated the documents. But I don't know how to invoke the when deprecation warnings adding the tests since we are just importing libraries. Do you have any suggestion?

Reply:
Also, do I need to add this warning in the files like `django/contrib/gis/db/backends/oracle/adapter.py`, `django/contrib/gis/db/backends/oracle/introspection.py` and `django/db/backends/oracle/introspection.py`?
Or only add it in `django/db/backends/oracle/base.py`?

Reply:
> Also, do I need to add this warning in the files like `django/contrib/gis/db/backends/oracle/adapter.py`, `django/contrib/gis/db/backends/oracle/introspection.py` and `django/db/backends/oracle/introspection.py`? Or only add it in `django/db/backends/oracle/base.py`?

Yes, we should have warnings in both backends `django.db.backends.oracle` and `django.contrib.gis.db.backends.oracle`. I'd prefer to have a single point (`base.py`) that will raise a warning
```python
try:
    import oracledb
except ImportError:
    import cx_Oracle as oracledb
    warnings.warn(
        "cx_Oracle is deprecated. Please use oracledb instead.",
        RemovedInDjango51Warning,
    )
```
In other imports, switching between libraries should be sufficient.

Reply:
This is still untested.